backend/tracim_backend/command/database.py

```diff
@@ -296,3 +301,154 @@ def take_action(self, parsed_args: argparse.Namespace) -> None:
                         engine.execute(
                             "ALTER INDEX {} RENAME TO {}".format(primary_key["name"], new_name)
                         )
+
+
+class MigrateStorageCommand(AppContextCommand):
+    auto_setup_context = False
+
+    def get_description(self) -> str:
+        return "migrate database storage"
+
+    def get_parser(self, prog_name: str) -> argparse.ArgumentParser:
+        parser = super().get_parser(prog_name)
+        parser.add_argument(
+            "-o",
+            "--old-config",
+            help="old configuration file to use",
+            dest="old_config_file",
+            required=True,
+        )
+        parser.add_argument(
+            "-f",
+            "--force-reupload",
+            help="force reupload all existing content to new storage",
+            dest="force",
+            default=False,
+            action="store_true",
+        )
+        return parser
+
+    def _reupload_depot_file(
```

Comment:
Wouldn't it be simpler if this function "only" converted an existing depot field to the new storage name?
Something like:
```python
def _get_new_storage_field(self, field_value, dest_storage, origin_storage, force_reupload):
   ...
  return FileIntent(file, field_value.filename, field_value.content_type),
```
Then use it like this:
```python
            for revision in revisions:
                logger.debug(
                    self, "migrate revision {} data to new storage".format(revision.revision_id)
                )
                revision.depot_file = self._get_new_storage_field(revision.depot_file, new_storage_name, original_storage_name, force_reupload)
            users = dbsession.query(User).all()
            for user in users:
                logger.debug(self, "migrate user {} data to new storage".format(user.user_id))
                user.avatar = self._get_new_storage_field(user.avatar, new_storage_name, original_storage_name, force_reupload)
                user.cropped_avatar = self._get_new_storage_field(user.cropped_avatar, new_storage_name, original_storage_name, force_reupload)
                user.cover = self._get_new_storage_field(user.cover, new_storage_name, original_storage_name, force_reupload)
                user.cropped_cover = self._get_new_storage_field(user.cropped_cover, new_storage_name, original_storage_name, force_reupload)

Reply:
it do like the idea.

Reply:
done.